File: ftc_app-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java
package org.firstinspires.ftc.teamcode;

import com.qualcomm.robotcore.hardware.HardwareMap;

abstract class Robot {
    abstract void setup(HardwareMap hardwareMap);
    abstract void setDrivePower(double power);
    abstract void setTurnPower(double power);
    abstract void turn(double power, double radians);
    abstract double getGyroHeading();
}

File: ftc_app-3.5/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/libraries/robot/Robot.kt
package org.firstinspires.ftc.teamcode.libraries.robot

import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode

open class Robot(val linearOpMode: LinearOpMode) {

    protected val components = hashMapOf<String, Component>()

    fun addComponent(component: Component, key: String) {
        components[key] = component
    }

}